services/data_integrity/sync_import_users.go

```diff
@@ -0,0 +1,244 @@
+
+	return nil
+}
+
+func removeDuplicateChannelMemberships(user *imports.UserImportData, flags SyncImportUsersFlags, logger *logrus.Logger) {
+	names := map[string]bool{}
+
+	teams := *user.Teams
+
+	chansOut := []imports.UserChannelImportData{}
+
+	for _, c := range *teams[0].Channels {
+		if names[*c.Name] {
+			logger.Warnf("Removing duplicate channel membership: user %s channel %s", *user.Username, *c.Name)
+		} else {
+			names[*c.Name] = true
+			chansOut = append(chansOut, c)
+		}
+	}
+
+	teams[0].Channels = &chansOut
+}
+
+func mergeImportFileUser(user *imports.UserImportData, flags SyncImportUsersFlags, client *model.Client4, logger *logrus.Logger) (usernameChanged, emailChanged bool, err error) {
+	usernameExists := false
+	emailExists := false
+
+	emailFromImport := strings.ToLower(*user.Email)
+	usernameFromImport := strings.ToLower(*user.Username)
+
+	existingUserByUsername, resp, err := client.GetUserByUsername(usernameFromImport, "")
+	if err != nil {
+		if resp == nil {
+			return false, false, errors.Wrap(err, "error fetching user by username")
+		}
+
+		if resp.StatusCode != 404 {
+			return false, false, err
+		}
+
+		logger.Debugf("Username %s does not exist in database", usernameFromImport)
+	} else {
+		usernameExists = true
+		logger.Debugf("Username %s exists in database", usernameFromImport)
+	}
+
+	existingUserByEmail, resp, err := client.GetUserByEmail(emailFromImport, "")
+	if err != nil {
+		if resp == nil {
+			return false, false, errors.Wrap(err, "error fetching user by email")
+		}
+
+		if resp.StatusCode != 404 {
+			return false, false, err
+		}
+
+		logger.Debugf("Email %s does not exist in database", emailFromImport)
+	} else {
+		emailExists = true
+		logger.Debugf("Email %s exists in database", emailFromImport)
+	}
+
+	if usernameExists && existingUserByUsername.Email != emailFromImport {
+		logger.Warnf("Username %s already exists, but has a different email. DB email: (%s) Import file email (%s)", usernameFromImport, existingUserByUsername.Email, emailFromImport)
+	}
+
+	if emailExists && existingUserByEmail.Username != usernameFromImport {
+		logger.Warnf("Email %s already exists, but has a different username. DB username: (%s) Import file username (%s)", emailFromImport, existingUserByEmail.Username, usernameFromImport)
+	}
+
+	avoidUpdatingEmail := false
+	avoidUpdatingUsername := false
+	if usernameExists && emailExists && existingUserByUsername.Id != existingUserByEmail.Id {
+		logger.Warnf("Found duplicate user %s in database. Import file username (%s). DB username 1 (%s). DB username 2 (%s). Import file email (%s). DB email 1 (%s). DB email 2 (%s)",
+			usernameFromImport, usernameFromImport, existingUserByUsername.Username, existingUserByEmail.Username, emailFromImport, existingUserByUsername.Email, existingUserByEmail.Email)
+
+		usernameUserActive := existingUserByUsername.DeleteAt == 0
+		emailUserActive := existingUserByEmail.DeleteAt == 0
+		if usernameUserActive && !emailUserActive {
```

Comment:
We can use the `Long` property of the command help to add an explanation of what the command is going to do on which cases, just to set the expectations of the user that is going to run it.